csharp/src/Apache.Arrow/Interfaces/IArrowArrayBuilder.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Apache.Arrow.Memory;
+using System;
+using System.Collections.Generic;
+
+namespace Apache.Arrow
+{
+    public interface IArrowArrayBuilder { }
```

Comment:
Empty interfaces should be avoided unless absolutely necessary. Is there something that can be added here?

[https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/interface](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/interface)

X AVOID using marker interfaces (interfaces with no members).